modules/account-lib/src/coin/hbar/transactionBuilder.ts

```diff
@@ -0,0 +1,145 @@
+import BigNumber from 'bignumber.js';
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import { BaseTransactionBuilder } from '../baseCoin';
+import {BuildTransactionError, ParseTransactionError} from '../baseCoin/errors';
+import { BaseAddress, BaseFee, BaseKey } from '../baseCoin/iface';
+import { proto } from '../../../resources/hbar/protobuf/hedera';
+import { Transaction } from './transaction';
+import { isValidAccount } from './utils';
+
+export class TransactionBuilder extends BaseTransactionBuilder {
+  protected _fee: BaseFee;
+  protected _transaction: Transaction;
+  protected _source: BaseAddress;
+  protected _startTime: proto.ITimestamp;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /**
+   * Set the transaction fees
+   *
+   * @param {BaseFee} fee The maximum gas to pay
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  fee(fee: BaseFee): this {
+    this.validateValue(new BigNumber(fee.fee));
+    this._fee = fee;
+    return this;
+  }
+
+  /**
+   * Set the transaction source
+   *
+   * @param {BaseAddress} address The source account
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  source(address: BaseAddress): this {
+    this.validateAddress(address);
+    this._source = address;
+    return this;
+  }
+
+  startTime(time: Date | string): this {
+    if (typeof time === 'string') {
+      const timeParts = time.split('.').map(v => +new BigNumber(v));
+      this._startTime = { seconds: timeParts[0], nanos: timeParts[1] };
+    } else {
+      this._startTime = { seconds: time.getTime() / 1000, nanos: 0 };
```

Comment:
so this gets weird with the resolution here. let's try to use this if we can:

`process.hrtime` for node
`performance.now` in browser

 https://stackoverflow.com/a/18197438 follow this basically

it's very possible this is a huge undertaking because of our env agnostic compilation. if this is really too difficult, let's circle back to this in a ticket. your call

Reply:
Done